### ISSUE TITLE
The type initializer for 'jdk.nashorn.internal.runtime.Timing' threw an exception.

### ISSUE THREAD

**Issue by eliyasbaby** [ikvm-revived/ikvm#24]
I have used [net_core_compat](https://github.com/ikvm-revived/ikvm/commits/net_core_compat) branch  and these [binaries](https://github.com/ams-ts-ikvm/ikvm-bin/tree/net_core_compat) by @ams-tschoening . because this has the fix for System.IO.FileStream construtor issue in .Net Core.
Now i am getting another exception with stack trace as belows.

> at jdk.nashorn.internal.runtime.ScriptEnvironment..ctor(Options options, PrintWriter out, PrintWriter err)
>    at jdk.nashorn.internal.runtime.Context..ctor(Options options, ErrorManager errors, PrintWriter out, PrintWriter err, ClassLoader appLoader, ClassFilter classFilter)
>    at jdk.nashorn.internal.runtime.Context..ctor(Options options, ErrorManager errors, ClassLoader appLoader, ClassFilter classFilter)
>    at jdk.nashorn.api.scripting.NashornScriptEngine.1.run()
>    at jdk.nashorn.api.scripting.NashornScriptEngine.1.run()
>    at java.security.AccessController.doPrivileged(Object , AccessControlContext , CallerID )
>    at java.security.AccessController.doPrivileged(PrivilegedAction action, AccessControlContext context, CallerID )
>    at jdk.nashorn.api.scripting.NashornScriptEngine..ctor(NashornScriptEngineFactory , String[] , ClassLoader , ClassFilter )
>    at jdk.nashorn.api.scripting.NashornScriptEngineFactory.getScriptEngine()
>    at javax.script.ScriptEngineManager.getEngineByName(String shortName)

Inner Exception and stack trace

> Method not found: 'System.Reflection.Emit.ModuleBuilder System.Reflection.Emit.AssemblyBuilder.DefineDynamicModule(System.String, System.String, Boolean)'.

 

>  at IKVM.Internal.DynamicClassLoader.CreateModuleBuilder(AssemblyName name)
>    at IKVM.Internal.DynamicClassLoader.CreateModuleBuilder()
>    at IKVM.Internal.DynamicClassLoader.Get(ClassLoaderWrapper loader)
>    at IKVM.Internal.ClassLoaderWrapper.GetTypeWrapperFactory()
>    at IKVM.Internal.ClassLoaderWrapper.DefineClassCritical(ClassFile f, ProtectionDomain protectionDomain)
>    at IKVM.Internal.ClassLoaderWrapper.DefineClass(ClassFile f, ProtectionDomain protectionDomain)
>    at Java_java_lang_ClassLoader.defineClass1(ClassLoader thisClassLoader, String name, Byte[] b, Int32 off, Int32 len, ProtectionDomain pd, String source)
>    at Java_java_lang_reflect_Proxy.defineClass0(ClassLoader classLoader, String name, Byte[] b, Int32 off, Int32 len)
>    at java.lang.reflect.Proxy.defineClass0(ClassLoader , String , Byte[] , Int32 , Int32 )
>    at java.lang.reflect.Proxy.access$400(ClassLoader , String , Byte[] , Int32 , Int32 )
>    at java.lang.reflect.Proxy.ProxyClassFactory.apply(ClassLoader , Class[] )
>    at java.lang.reflect.Proxy.ProxyClassFactory.apply(Object , Object )
>    at java.lang.reflect.WeakCache.Factory.get()
>    at java.lang.reflect.WeakCache.get(Object , Object )
>    at java.lang.reflect.Proxy.getProxyClass0(ClassLoader , Class[] )
>    at java.lang.reflect.Proxy.newProxyInstance(ClassLoader loader, Class[] interfaces, InvocationHandler h, CallerID )
>    at sun.reflect.annotation.AnnotationParser.1.run()
>    at sun.reflect.annotation.AnnotationParser.1.run()
>    at java.security.AccessController.doPrivileged(Object , AccessControlContext , CallerID )
>    at java.security.AccessController.doPrivileged(PrivilegedAction action, CallerID )
>    at sun.reflect.annotation.AnnotationParser.annotationForMap(Class type, Map memberValues)
>    at Java_java_lang_Class.FreezeOrWrapAttribute(Annotation ann)
>    at Java_java_lang_Class.AnnotationsToMap(ClassLoaderWrapper loader, Object[] objAnn)
>    at Java_java_lang_Class.getDeclaredAnnotationsImpl(Class thisClass)
>    at java.lang.Class.createAnnotationData(Int32 )
>    at java.lang.Class.annotationData()
>    at java.lang.Class.getAnnotation(Class annotationClass)
>    at jdk.nashorn.internal.runtime.Timing..cctor()
   
I am using .Net core-3.1. i have tried installing System.Reflection nuget,but that did not fix the problem.

**Comment by ams-tschoening** [comment]
`net_core_compat` might be misleading, it's only what I needed to make things work in the context of my project. It's not meant to be really compatible with .NET Core, so things are expected to NOT work in all environments. From my point of view you have two choices: Fix the concrete problem on your own, most likely by commenting the problematic code. If you have a look at the history of the mentioned branch, this is mostly what I did as well.

Or you have a look at newer approaches:

https://github.com/ikvm-revived/ikvm/issues/20#issuecomment-668110864
